pygsti/construction/modelconstruction.py

```diff
@@ -1254,52 +1253,58 @@ def create_crosstalk_free_model(num_qubits, gate_names, nonstd_gate_unitaries={}
     else:
         global_idle_op = None
 
+    # TODO: While TensorProdSPAMVec does not work with ComposedSPAMVec, build up prep/povm as n-qubit ops first    
     prep_layers = {}
-    if evotype == 'chp':
-        assert ('prep' not in all_keys), "Cannot specify 'prep' error specification with CHP evotype"
-
-        # For CHP, prep can only be idle gate
-        # TODO: This should be replaced by a SPAMVec in the near future
-        # Only works because LocalNoiseModel allows it, which it probably shouldn't
-        rho0 = _op.ComposedOp([_op.EmbeddedOp(range(num_qubits), [i], _op.StaticStandardOp('Gi', 'chp'))
-                               for i in range(num_qubits)])
-        prep_layers['rho0'] = rho0
-    elif 'prep' in all_keys:
-        if 'prep' in depolarization_strengths and depolarization_parameterization != 'lindblad':
-            _warnings.warn(("'prep' error specification requires Lindblad parameterization, "
-                           "depolarization parameterization '%s' overridden!" % depolarization_parameterization))
-        elif 'prep' in stochastic_error_probs and stochastic_parameterization != 'lindblad':
-            _warnings.warn(("'prep' error specification requires Lindblad parameterization, "
-                           "stochastic parameterization '%s' overridden!" % stochastic_parameterization))
-
-        rho_base1Q = _spamvec.ComputationalSPAMVec([0], evotype, 'prep')
-        # Override parameterization to force Lindblad
-        err_gate = _get_error_gate('prep', _op.StaticStandardOp('Gi', evotype), depolarization_strengths,
-                                   stochastic_error_probs, lindblad_error_coeffs,
-                                   "lindblad", "lindblad", "lindblad")
-        prep1Q = _spamvec.LindbladSPAMVec(rho_base1Q, err_gate, 'prep')
-        prep_factors = [prep1Q.copy() for i in range(num_qubits)] if independent_gates else [prep1Q] * num_qubits
-        prep_layers['rho0'] = _spamvec.TensorProdSPAMVec('prep', prep_factors)
+    if 'prep' in all_keys:
+        # rho_base1Q = _spamvec.ComputationalSPAMVec([0], evotype, 'prep')
+        # err_gate = _get_error_gate('prep', _op.StaticStandardOp('Gi', evotype), depolarization_strengths,
+        #                            stochastic_error_probs, lindblad_error_coeffs,
+        #                            depolarization_parameterization, stochastic_parameterization,
+        #                            lindblad_parameterization)
+        # if isinstance(err_gate, _op.LindbladOp):
+        #     prep1Q = _spamvec.LindbladSPAMVec(rho_base1Q, err_gate, 'prep')
+        # else:
+        #     prep1Q = ComposedSPAMVec(rho_base1Q, err_gate, 'prep')
+        # prep_factors = [prep1Q.copy() for i in range(num_qubits)] if independent_gates else [prep1Q] * num_qubits
+        # prep_layers['rho0'] = _spamvec.TensorProdSPAMVec('prep', prep_factors)
+        rho_baseNQ = _spamvec.ComputationalSPAMVec([0,]*num_qubits, evotype, 'prep')
```

Comment:
Do we want to still have the option of having a TensorProdSPAMVec here?  It's probably more performant than a composition of embedded ops, and maybe should be used in cases when it works and when the user desires.  I think it should work whenever the factor spam vectors have a definite num_params, and we can probably fix to work in general.

Reply:
I originally just intended this to be a temporary measure, but in hindsight, it's probably a good call for everything coming into develop to have good default behavior.

I've now added a `tensorprod_spamvec_povm` kwarg, controlling which construction is used. The default is the old TensorProd construction, and I can just use False in my runs until we get a proper fix for the TensorProd variants.